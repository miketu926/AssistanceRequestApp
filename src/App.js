import React from 'react';

import Form from './component/form';

function App() {

  return (
    <div className='main-box'>
      <Form />
    </div>
  );
}

export default App;
