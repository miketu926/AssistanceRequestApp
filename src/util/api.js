// update getURL and postURL
let getURL = 'http://192.168.99.101:49567/api/service-types';
let postURL = 'http://192.168.99.101:49567/api/assistance-requests';
// let getURL = 'http://localhost:49567/api/service-types';
// let postURL = 'http://localhost:49567/api/assistance-requests';

export const getServiceTypes = () => {
  return fetch(getURL);
};

export const postAssistanceRequest = (data) => {
  return fetch(postURL, {
    method: 'POST',
    body: JSON.stringify(data),
  });
};