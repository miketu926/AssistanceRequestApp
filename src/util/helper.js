export const stateToJson = (state) => {
  return {
    assistance_request: {
      contact: {
        first_name: state.firstName,
        last_name: state.lastName,
        email: state.email,
      },
      service_type: state.serviceType,
      description: state.description,
    }
  };
}
