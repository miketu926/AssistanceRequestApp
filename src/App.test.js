import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { getServiceTypes, postAssistanceRequest } from './util/api';
import { mount, configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Form from './component/form';

configure({ adapter: new Adapter() });

var getResponse = {
  "data": [
    {
      "display_name": "Benefits",
      "id": "benefits"
    },
    {
      "display_name": "Employment",
      "id": "employment"
    },
    {
      "display_name": "Healthcare",
      "id": "healthcare"
    },
    {
      "display_name": "Housing",
      "id": "housing"
    },
    {
      "display_name": "Legal",
      "id": "legal"
    }
  ]
};

var postSample = {
  assistance_request: {
    contact: {
      first_name: "Mike",
      last_name: "Tu",
      email: "miketu926@gmail.com",
    },
    service_type: "Interview",
    description: "Hoping to add value to the engineering team at UniteUs to further the company's growth",
  }
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('the API', () => {
  it('performs a GET request and returns a 200 status', () => {
    return getServiceTypes().then(res => {
      expect(res.status).toEqual(200);
    });
  });
  it('performs a GET request and returns a json', () => {
    return getServiceTypes().then(res => res.json()).then(res => {
      expect(res.data).toEqual(getResponse.data);
    });
  });
  it('performs a POST request and returns one of the 4 statuses', () => {
    return postAssistanceRequest(postSample).then(res => {
      expect(res.status === 201 || res.status === 401 || res.status === 500 || res.status === 503).toBeTruthy();
    });
  });
});

describe('the Form', () => {
  const state = {
    firstName: "Mike",
    lastName: "Tu",
    email: "miketu926@gmail.com",
    serviceType: "Interview",
    description: "Time to shine!",
    checked: true,
  };

  it('should match its empty snapshot', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls handleSubmit when the form is filled and submitted', () => {
    const component = shallow(<Form />);
    const preventDefault = jest.fn();

    component.setState({
      state
    });

    component.find('form').simulate('submit', { preventDefault });
    expect(component).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
  });



});