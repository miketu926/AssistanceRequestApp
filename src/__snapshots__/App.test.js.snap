// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Form calls handleSubmit when the form is filled and submitted 1`] = `ShallowWrapper {}`;

exports[`the Form calls handleSubmit when the form is submitted 1`] = `ShallowWrapper {}`;

exports[`the Form should match its empty snapshot 1`] = `
<form
  className="col w-80"
  onSubmit={[Function]}
>
  <h3
    className="col mb-4 border-bottom pb-1"
  >
    New Assistance Request
  </h3>
  <input
    className="col"
    onChange={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />
  <div
    className="col text-danger text-right small mb-2"
  >
    required
  </div>
  <input
    className="col"
    onChange={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />
  <div
    className="col text-danger text-right small mb-2"
  >
    required
  </div>
  <input
    className="col"
    onChange={[Function]}
    placeholder="Email Address"
    type="email"
    value=""
  />
  <div
    className="col text-danger text-right small mb-2"
  >
    required
  </div>
  <select
    className="col"
    id="serviceType"
    onChange={[Function]}
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      Select Service Type
    </option>
  </select>
  <div
    className="col text-danger text-right small mb-2"
  >
    required
  </div>
  <textarea
    className="col"
    cols="60"
    onChange={[Function]}
    placeholder="Description"
    rows="7"
    value=""
  />
  <div
    className="col mt-2 mb-4"
  >
    <input
      className="col-1"
      id="checkbox"
      name="checkbox"
      onClick={[Function]}
      type="checkbox"
    />
    <small
      className="col-11"
      htmlFor="checkbox"
    >
      I hereby accept the terms of service for THE NETWORK and the Privacy Policy.
    </small>
  </div>
  <input
    className="col-4 offset-8"
    disabled={true}
    type="submit"
    value="Get Assistance"
  />
</form>
`;
